src/psock.c

```diff
@@ -0,0 +1,132 @@
+/*
```

Comment:
Rename this file to `tcp_psock.h` and `tcp_psock.c`, just to emphasize that they are extensions to `tcp.{h.c}`.